docs/usage/key-concepts/how-renovate-works.md

```diff
@@ -0,0 +1,113 @@
+- [manager](../modules/manager/index.md)
+- [platform](../modules/platform/index.md)
+- [versioning](../modules/versioning.md).
+
+Renovate uses these modules in order:
+
+1. The platform module is used to interact with the source control platform and clone the repository
+1. The manager module looks for files based on their name and extracts the dependencies (each dependency has a datasource)
+1. The datasource module looks up versions of the dependency
+1. The versioning module validates and sorts the returned versions
+
+For example:
+
+1. The `gitlabci` manager finds a dependency: `python:3.10-alpine` which has the `docker` datasource
+2. The `docker` datasource looks for versions and finds: `[python:3.9,python:3.9-alpine,python:3.10,python:3.10-alpine,python:3.11,python:3.11-alpine]`
+3. The `docker` versioning returns `python:3.11-alpine`, because that version is compatible with `python:3.10-alpine`
+
+# Workflow
+
+Here's an overview of the workflow:
+
+```mermaid
+flowchart TB
+    subgraph INITIALIZATION
+        direction TB
+        MC[Merge configurations \n most important to least: \n cli > env > file > default]
+        MC --> IP[Initialize platform]
+        IP --> AD[Query the platform for repositories]
+        AD --> NFIL[Narrow the list with filters]
+    end
+
+    subgraph REPOSITORY
+       direction TB
+       FER[[For each repository]]
+
+      subgraph EXTRACTD[EXTRACT DEPENDENCIES]
+          direction TB
+          CLBRANCH[Extract base branches]
+          CLBRANCH --> VULN[Check for vulnerabilities]
+          VULN --> CC[[For each manager]]
+          CC -->|managerA| CD["..."]
+          CC -->|managerB| CCF["match files"]
+          CCF --> CFEF[[For each file]]
+          CFEF -->|file1| CCD1[Extract dependency]
+          CFEF -->|file2| CCD2[...]
+      end
+
+      subgraph LOOKUP[LOOK UP UPDATES]
+          direction TB
+          UC[[For each manager]]
+          UC -->|managerA| UD["..."]
+          UC -->|managerB| UFEF[[For each file]]
+          UFEF -->|file1| FED[[For each dependency]]
+          UFEF -->|file2| FED2[...]
+          FED -->|dep1| D1[...]
+          D1 -..-> CU
+          FED -->|dep2| D2[use datasource to \n fetch versions]
+          D2 --> J[use versioning to find \n next valid update]
+          FED2 -...-> CU
+          UD -....-> CU
+          J --> CU[Look up updates]
+      end
+
+      subgraph WRITEU[WRITE UPDATES]
+        direction TB
+        FEU[[For each update]]
+        FEU --> AUCOND[Check if branch needed: \n existing/rebase/concurrent amount]
+        AUCOND --> AU[Create branch\nApply update\nCreate PR]
+      end
+
+      subgraph FINALIZE[FINALIZE]
+      direction TB
+        CM[Check for config migration]
+        CM --> CSB[Clean stale branches]
+
+      end
+
+      FER --> IRPO[Initialize repository]
+
+      IRPO --> EXTRACTD
+      EXTRACTD --> LOOKUP
+
+      LOOKUP --> WRITEU
+
+      WRITEU --> FINALIZE
+
+    end
+
+    INITIALIZATION --> REPOSITORY
```

Comment:
```suggestion
flowchart TB
    subgraph INITIALIZATION
        direction TB
        MC[Merge configurations \n most important to least: \n cli > env > file > default]
        MC --> IP[Initialize platform]
        IP --> AD[Query the platform for repositories]
        AD --> NFIL[Narrow the list with filters]
    end

    subgraph REPOSITORY
       direction TB
       FER[For each repository]

      subgraph EXTRACTD[EXTRACT DEPENDENCIES]
          direction TB
          CLBRANCH[Extract base branches]
          CLBRANCH --> VULN[Check for vulnerabilities]
          VULN --> CC[For each manager]
          CC -->|managerA| CD["..."]
          CC -->|managerB| CCF["match files"]
          CCF --> CFEF[For each file]
          CFEF -->|file1| CCD1[Extract dependency]
          CFEF -->|file2| CCD2[...]
      end

      subgraph LOOKUP[LOOK UP UPDATES]
          direction TB
          UC[For each manager]
          UC -->|managerA| UD["..."]
          UC -->|managerB| UFEF[For each file]
          UFEF -->|file1| FED[For each dependency]
          UFEF -->|file2| FED2[...]
          FED -->|dep1| D1[...]
          D1 -..-> CU
          FED -->|dep2| D2[use datasource to \n fetch versions]
          D2 --> J[use versioning to find \n next valid update]
          FED2 -...-> CU
          UD -....-> CU
          J --> CU[Look up updates]
      end

      subgraph WRITEU[WRITE UPDATES]
        direction TB
        FEU[For each update]
        FEU --> AUCOND[Check if branch needed: \n existing/rebase/concurrent amount]
        AUCOND --> AU[Create branch\nApply update\nCreate PR]
      end

      subgraph FINALIZE[FINALIZE]
      direction TB
        CM[Check for config migration]
        CM --> CSB[Clean stale branches]

      end

      FER --> IRPO[Initialize repository]

      IRPO --> EXTRACTD
      EXTRACTD --> LOOKUP

      LOOKUP --> WRITEU

      WRITEU --> FINALIZE

    end

    INITIALIZATION --> REPOSITORY
```

Right now some boxes have a extra set of `[]` characters, like this: `[[For each ...]]`. I think it looks cleaner without the extra borders, so like this: `[For each ...]`.

This suggestion makes all the boxes look the same.

Reply:
It isn't "some box" but the boxes that mean "iteration over" 

You give through the simple boxes but the [[]] boxes parallelize the actions , I thought I was better to differentiate , let me know what you think about this

Reply:
> You give through the simple boxes but the [[]] boxes parallelize the actions , I thought I was better to differentiate , let me know what you think about this

You're right that we should differentiate the boxes. But the `[[]]` box does not _feel_ very different to me.

The [Mermaid.js Flowcharts syntax, node shapes](https://mermaid.js.org/syntax/flowchart.html#node-shapes) shows examples of different node shapes. Can you try the [hexagon node](https://mermaid.js.org/syntax/flowchart.html#a-hexagon-node)? That feels different enough, and it's not too distracting, I hope. 😄

Reply:
Here it is https://github.com/azro352/renovate/blob/feat/how-does-work-basic/docs/usage/key-concepts/how-renovate-works.md